contracts/contracts/coordination/Coordinator.sol

```diff
@@ -45,26 +48,44 @@ contract Coordinator is Ownable {
     // TODO: Optimize layout
     struct Ritual {
         address initiator;
-        uint32 dkgSize;
         uint32 initTimestamp;
-        uint32 totalTranscripts;
-        uint32 totalAggregations;
-        BLS12381.G1Point publicKey;
+        uint32 endTimestamp;
+        uint16 totalTranscripts;
+        uint16 totalAggregations;
+        address authority;
+        uint16 dkgSize;
         bool aggregationMismatch;
+        BLS12381.G1Point publicKey;
         bytes aggregatedTranscript;
         Participant[] participant;
     }
 
-    Ritual[] public rituals;
+    bytes32 public constant INITIATOR_ROLE = keccak256("INITIATOR_ROLE");
 
     IAccessControlApplication public immutable application;
-    uint32 public timeout;
-    uint32 public maxDkgSize;
 
-    constructor(IAccessControlApplication app, uint32 _timeout, uint32 _maxDkgSize) {
-        application = app;
+    Ritual[] public rituals;
+    uint32 public timeout;
+    uint16 public maxDkgSize;
+    bool public isInitiationPublic;
+    IFeeModel feeModel;  // TODO: Consider making feeModel specific to each ritual
+    IReimbursementPool reimbursementPool;
+    uint256 public totalPendingFees;
+    mapping(uint256 => uint256) public pendingFees;
+
+    constructor(
+        IAccessControlApplication _stakes,
+        uint32 _timeout,
+        uint16 _maxDkgSize,
+        address _admin,
+        address _feeModel
+    ) AccessControlDefaultAdminRules(0, _admin)
+    {
+        require(address(_feeModel.stakes()) == address(_stakes), "Invalid stakes for fee model");
+        application = _stakes;
         timeout = _timeout;
         maxDkgSize = _maxDkgSize;
+        feeModel = IFeeModel(_feeModel);
```

Comment:
Shall we permit the `feeModel` address to be updated by the proper role?

Reply:
Yeah, it's possible, but I'd rather handle that in a later PR, when we decide how do we want to manage fee models (see comment in L71). For our immediate needs (devnets), I don't foresee we will want to change the fee model